src/Streamly/Internal/FileSystem/Event/Windows.hs

```diff
@@ -279,27 +278,28 @@ getConfigRecMode Config{..} = watchRec
 
 data Event = Event
     { eventFlags :: DWORD
-    , eventRelPath :: String
-    , eventRootPath :: String
+    , eventRelPath :: Path
+    , eventRootPath :: Path
     , totalBytes :: DWORD
-    } deriving (Show, Ord, Eq)
+    }
 
 -- For reference documentation see:
 --
 -- See https://docs.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-file_notify_information
 data FILE_NOTIFY_INFORMATION = FILE_NOTIFY_INFORMATION
     { fniNextEntryOffset :: DWORD
     , fniAction :: DWORD
-    , fniFileName :: String
-    } deriving Show
+    , fniFileName :: Path
+    }
 
 type LPOVERLAPPED_COMPLETION_ROUTINE =
     FunPtr ((DWORD, DWORD, LPOVERLAPPED) -> IO ())
 
 -- | A handle for a watch.
-getWatchHandle :: FilePath -> IO (HANDLE, FilePath)
+getWatchHandle :: Path -> IO (HANDLE, Path)
 getWatchHandle dir = do
-    h <- createFile dir
+    let dirStr = Path.toString dir
```

Comment:
We should not use `Path.toString`. Should use a version of createFile that takes the raw string.